### PR TITLE
[CDAP-7602] Feature/ui wrangler 3

### PR DESCRIPTION
**Pull request by elfenheart**
known bug:
- Action dropdown will go under the left panel
- Action buttons are difficult to press
- No tooltip for actions
- No history styling

### CODE REVIEW

File: cdap-ui/app/cdap/components/Wrangler/Redux/WranglerStore.js [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    history: [],
+    histogram: {},
+    columnTypes: {},
+    filter: null,
+    sort: null,
+    sortAscending: true
+  }
+};
+
+const wrangler = (state = defaultAction, action = defaultInitialState) => {
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/Redux/WranglerStore.js]: should this be (state = defaultInitialState, action = defaultAction) ?

File: cdap-ui/app/cdap/components/Wrangler/Redux/WranglerStore.js [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    columnTypes,
+    histogram,
+    errors
+  };
+}
+
+const WranglerStoreWrapper = () => {
+  return createStore(
+    combineReducers({
+      wrangler,
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/Redux/WranglerStore.js]: unnecessary comma?

File: cdap-ui/app/cdap/components/Wrangler/index.js [hunk 3]
```diff
@@ -97,78 +122,108 @@ export default class Wrangler extends Component {
...
+            <i className="fa fa-plus-circle"></i>
           </div>
-          <div className="col-xs-6">
-            <h3>Copy Input Text</h3>
-            <textarea
-              className="form-control"
-              onChange={this.handleTextInput}
-            />
+          <div className="plus-button-helper-text">
+            <h4>Copy-paste data here</h4>
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/index.js]: i18n?

File: cdap-ui/app/cdap/components/Wrangler/index.js [hunk 4]
```diff
@@ -97,78 +122,108 @@ export default class Wrangler extends Component {
...
+      return null;
+    }
+
+    if (this.state.loading) {
+      return (
+        <div className="loading text-center">
+          <div>
+            <span className="fa fa-spinner fa-spin"></span>
+          </div>
+          <h3>Parsing...</h3>
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/index.js]: i18n

File: cdap-ui/app/cdap/components/Wrangler/index.js [hunk 5]
```diff
@@ -97,78 +122,108 @@ export default class Wrangler extends Component {
...
+                  /> First line as column name
+                </label>
+              </div>
+
+              <div className="checkbox">
+                {/* skipEmptyLines */}
+                <label>
+                  <input type="checkbox"
+                    onChange={this.handleSetSkipEmptyLines}
+                  /> Skip empty lines
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/index.js]: i18n

File: cdap-ui/app/cdap/components/Wrangler/index.js [hunk 6]
```diff
@@ -97,78 +122,108 @@ export default class Wrangler extends Component {
...
 
         <br/>
+
         <div className="text-center">
           <button
             className="btn btn-primary"
-            onClick={this.wrangle}
+            onClick={this.onWrangleClick}
           >
             Wrangle
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/index.js]: i18n here too

File: cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js [hunk 7]
```diff
@@ -16,647 +16,355 @@
...
-                            key={shortid.generate()}
-                            className={classnames({
-                              active: this.state.activeSelection === head
-                            })}
+
+        <div className="wrangle-results">
+          <div className="wrangler-data-metrics">
+            <div className="metric-block">
+              <h3 className="text-success">{this.state.data.length}</h3>
+              <h5>Rows</h5>
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js]: i18n

File: cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js [hunk 8]
```diff
@@ -16,647 +16,355 @@
...
+            </div>
+
+            <div className="metric-block">
+              <h3 className="text-success">{this.state.headersList.length}</h3>
+              <h5>Columns</h5>
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js]: i18n

File: cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js [hunk 9]
```diff
@@ -16,647 +16,355 @@
...
+            </div>
+
+            <div className="metric-block">
+              <h3 className="text-danger">{errorCount}</h3>
+              <h5>Errors</h5>
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js]: i18n

File: cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js [hunk 10]
```diff
@@ -16,647 +16,355 @@
...
+          <Filter column={this.state.activeSelection} />
+
+          <div
+            className="transform-item"
+            onClick={this.onHistogramDisplayClick}
+          >
+            <span className="fa fa-bar-chart"></span>
+            <span className="transform-item-text">
+              <span>{ this.state.showHistogram ? 'Hide' : 'Show'}</span>
+              <span>Histogram</span>
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js]: i18n

File: cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js [hunk 11]
```diff
@@ -16,647 +16,355 @@
...
-        formattedData = titlecaseColumn(this.state.data, columnToTransform);
-        break;
+  render() {
+    if (this.state.loading) {
+      return (
+        <div className="loading text-center">
+          <div>
+            <span className="fa fa-spinner fa-spin"></span>
+          </div>
+          <h3>Wrangling...</h3>
```
Comment (PGuimarais) [inline cdap-ui/app/cdap/components/Wrangler/WrangleData/index.js]: i18n

**Comment by PGuimarais** [comment]
Need to add i18n to rendered text. It needs to be applied to:

columnactions/substringaction.js
columnactions/splitaction.js
columnactions/renameaction.js
columnactions/mergeactions.js
wrangler/wrangledata/index.js 
wrangler/index.js

**Comment by elfenheart** [comment]
@PGuimarais i18n is not scoped out for this yet, there are still quite a bit of changes coming, so that will come later.

**Comment by ajainarayanan** [comment]
Minor things,

1. We use `shortid.generate()` in render function and that will cause issues with repeated rendering.
2. History if big makes the close button overflow to the next line. 
3. Redux folder under Wrangler could be renamed to Store. 

But these are not blockers for merging this PR 👍 ⚡️

**Comment by elfenheart** [comment]
PR squashed. Merging

**Pull request merged**